cnf-certification-test/performance/suite.go

```diff
@@ -233,6 +233,30 @@ func testSchedulingPolicyInCPUPool(env *provider.TestEnvironment,
 	testhelper.AddTestResultReason(compliantContainersPids, nonCompliantContainersPids, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 
+func getExecProbesCmds(c *provider.Container) map[string]bool {
+	cmds := map[string]bool{}
+
+	if c.LivenessProbe != nil && c.LivenessProbe.Exec != nil {
+		cmd := strings.Join(c.LivenessProbe.Exec.Command, "")
+		cmd = strings.Join(strings.Fields(cmd), "")
+		cmds[cmd] = true
+	}
+
+	if c.ReadinessProbe != nil && c.ReadinessProbe.Exec != nil {
+		cmd := strings.Join(c.ReadinessProbe.Exec.Command, "")
+		cmd = strings.Join(strings.Fields(cmd), "")
+		cmds[cmd] = true
+	}
+
+	if c.StartupProbe != nil && c.StartupProbe.Exec.Command != nil {
+		cmd := strings.Join(c.StartupProbe.Exec.Command, "")
+		cmd = strings.Join(strings.Fields(cmd), "")
+		cmds[cmd] = true
+	}
+
+	return cmds
+}
```

Comment:
```suggestion
func getExecProbesCmds(c *provider.Container) map[string]bool {
    cmds := make(map[string]bool)

    probeCommands := []func() []string{
        func() []string { return c.LivenessProbe.Exec.Command },
        func() []string { return c.ReadinessProbe.Exec.Command },
        func() []string { return c.StartupProbe.Exec.Command },
    }

    for _, cmdFunc := range probeCommands {
        if cmd := cmdFunc(); cmd != nil {
            cleanedCmd := strings.Join(cmd, "")
            cleanedCmd = strings.Join(strings.Fields(cleanedCmd), "")
            cmds[cleanedCmd] = true
        }
    }

    return cmds
}
```

Reply:
[Clear is better than clever](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=875s) :)

Reply:
@rdavid let keep it as-is this time. I adapted you idea to match the current logic and I also prefer it as it is now. See:
func getExecProbesCmds(c *provider.Container) map[string]bool {
	cmds := make(map[string]bool)
	probeCommands := []func() *v1.Probe{
		func() *v1.Probe { return c.LivenessProbe },
		func() *v1.Probe { return c.ReadinessProbe },
		func() *v1.Probe { return c.StartupProbe },
	}
	for _, cmdFunc := range probeCommands {
		if probe := cmdFunc(); probe != nil {
			if probe.Exec != nil {
				cleanedCmd := strings.Join(probe.Exec.Command, "")
				cleanedCmd = strings.Join(strings.Fields(cleanedCmd), "")
				cmds[cleanedCmd] = true
			}
		}
	}
	return cmds
}